stablehlo/tests/interpret_multiply.mlir

```diff
@@ -0,0 +1,232 @@
+  %1 = stablehlo.constant dense<[-32768, -1, 128, -129, 32767]> : tensor<5xi16>
+  %2 = stablehlo.multiply %0, %1 : tensor<5xi16>
+  func.return %2 : tensor<5xi16>
+  // CHECK-NEXT: tensor<5xi16>
+  // CHECK-NEXT: 0 : i16
+  // CHECK-NEXT: -1 : i16
+  // CHECK-NEXT: 16384 : i16
+  // CHECK-NEXT: 16641 : i16
+  // CHECK-NEXT: 0 : i16
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: mul_op_test_ui16
+func.func @mul_op_test_ui16() -> tensor<2xui16> {
+  %0 = stablehlo.constant dense<[0, 256]> : tensor<2xui16>
+  %1 = stablehlo.constant dense<[65535, 256]> : tensor<2xui16>
+  %2 = stablehlo.multiply %0, %1 : tensor<2xui16>
+  func.return %2 : tensor<2xui16>
+  // CHECK-NEXT: tensor<2xui16>
+  // CHECK-NEXT: 0 : ui16
+  // CHECK-NEXT: 0 : ui16
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: mul_op_test_si32
+func.func @mul_op_test_si32() -> tensor<5xi32> {
+  %0 = stablehlo.constant dense<[0, 1, 32768, -32769, 0]> : tensor<5xi32>
+  %1 = stablehlo.constant dense<[-2147483648, -1, 32768, -32769, 2147483647]> : tensor<5xi32>
+  %2 = stablehlo.multiply %0, %1 : tensor<5xi32>
+  func.return %2 : tensor<5xi32>
+  // CHECK-NEXT: tensor<5xi32>
+  // CHECK-NEXT: 0 : i32
+  // CHECK-NEXT: -1 : i32
+  // CHECK-NEXT: 1073741824 : i32
+  // CHECK-NEXT: 1073807361 : i32
+  // CHECK-NEXT: 0 : i32
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: mul_op_test_ui32
+func.func @mul_op_test_ui32() -> tensor<2xui32> {
+  %0 = stablehlo.constant dense<[0, 65536]> : tensor<2xui32>
+  %1 = stablehlo.constant dense<[4294967295, 65536]> : tensor<2xui32>
+  %2 = stablehlo.multiply %0, %1 : tensor<2xui32>
+  func.return %2 : tensor<2xui32>
+  // CHECK-NEXT: tensor<2xui32>
+  // CHECK-NEXT: 0 : ui32
+  // CHECK-NEXT: 0 : ui32
+}
+
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: mul_op_test_si64
+func.func @mul_op_test_si64() -> tensor<5xi64> {
+  %0 = stablehlo.constant dense<[0, 1, 2147483648, -2147483649, 0]> : tensor<5xi64>
+  %1 = stablehlo.constant dense<[-9223372036854775808, -1, 2147483648, -2147483649, 9223372036854775807]> : tensor<5xi64>
+  %2 = stablehlo.multiply %0, %1 : tensor<5xi64>
+  func.return %2 : tensor<5xi64>
+  // CHECK-NEXT: tensor<5xi64>
+  // CHECK-NEXT: 0 : i64
+  // CHECK-NEXT: -1 : i64
+  // CHECK-NEXT: 4611686018427387904 : i64
+  // CHECK-NEXT: 4611686022722355201 : i64
+  // CHECK-NEXT: 0 : i64
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: mul_op_test_ui64
+func.func @mul_op_test_ui64() -> tensor<2xui64> {
+  %0 = stablehlo.constant dense<[0, 4294967296]> : tensor<2xui64>
+  %1 = stablehlo.constant dense<[18446744073709551615, 4294967296]> : tensor<2xui64>
+  %2 = stablehlo.multiply %0, %1 : tensor<2xui64>
+  func.return %2 : tensor<2xui64>
+  // CHECK-NEXT: tensor<2xui64>
+  // CHECK-NEXT: 0 : ui64
+  // CHECK-NEXT: 0 : ui64
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: mul_op_test_c64
```

Comment:
Let's reorder the tests to match `spec_draft.md`'s order, and `bf16` should come first then `f16` https://github.com/openxla/stablehlo/blob/c3588936a6554e5ec5e167393a7e658fdcae0206/docs/spec_draft.md?plain=1#L7-L28